assets/js/__tests__/span-button.test.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * @jest-environment jsdom
+ */
+import addSpanEventHandlers from "../span-button";
+
+describe("addSpanEventHandlers", () => {
+  beforeEach(() => {
+    document.body.innerHTML = `
+    <div>
+      <span id="span" role="button">span button<span>
+    </div>
+    `;
+  });
+
+  test("makes span button click on enter key", () => {
+    const stub = jest.fn();
+    const spanEl = document.getElementById("span");
+    spanEl.onclick = stub;
+    const enterKeyEvent = new KeyboardEvent("keydown", { key: "Enter" });
+
+    addSpanEventHandlers();
+    spanEl.dispatchEvent(enterKeyEvent);
+
+    expect(stub).toBeCalled();
+  });
+
+  test("makes span button click on spacebar key", () => {
+    const stub = jest.fn();
+    const spanEl = document.getElementById("span");
+    spanEl.onclick = stub;
+    const spaceBarKeyEvent = new KeyboardEvent("keydown", { key: "Spacebar" });
```

Comment:
I would have assumed that spacebar keydown events are always dispatched with the same value (in my browser it's `" "`). Is this not the case due to i18n/browser implementations/other reasons?

Reply:
Some older browsers apparently use "Spacebar" so it's here for backwards compatibility/completeness. Probably not totally necessary, but why not since it's easy:

https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key/Key_Values
> [1] Older browsers may return "Spacebar" instead of " " for the Space Bar key. Firefox did so until version 37, as did Internet Explorer 9, 10, and 11.

Reply:
Good to know, thanks!